source/beamline/REIXS/REIXSSampleMotor.cpp

```diff
@@ -0,0 +1,377 @@
+		                           AMControl* verticalRotationControl,
+		                           QObject *parent,
+		                           const QString& description) :
+
+    AMPseudoMotorControl(name, units, parent, description)
+{
+	direction_ = direction;
+
+	if(direction_ == AMMotorGroupObject::VerticalMotion) {
+
+		AMErrorMon::alert(this,
+		                  REIXS_SAMPLE_MOTOR_INVALID_DIRECTION,
+		                  QString("REIXS Sample Motor initialized in vertical axis"));
+	}
+
+	horizontalTranslationControl_ = horizontalTranslationControl;
+	normalTranslationControl_ = normalTranslationControl;
+	verticalRotationControl_  = verticalRotationControl;
+
+	angleOffset_ = 0;
+	connectedOnce_ = false;
+
+	addChildControl(horizontalTranslationControl_);
+	addChildControl(normalTranslationControl_);
+	addChildControl(verticalRotationControl_);
+
+	connect(verticalRotationControl_, SIGNAL(valueChanged(double)),
+	        this, SLOT(onVerticalAxisRotated(double)));
+}
+
+
+double REIXSSampleMotor::angleOffset() const
+{
+	return angleOffset_;
+}
+
+double REIXSSampleMotor::totalRotation() const
+{
+	if(connected_) {
+
+		double combinedRotation = verticalRotationControl_->value() + angleOffset_;
+
+		while(combinedRotation > 360) {
+			combinedRotation -= 360;
+		}
+
+		while(combinedRotation < 0) {
+			combinedRotation += 360;
+		}
+
+		return combinedRotation;
+	}
+
+	return 0;
+}
+
+
+AMMotorGroupObject::MotionDirection REIXSSampleMotor::direction() const
+{
+	return direction_;
+}
+
+bool REIXSSampleMotor::canMeasure() const
+{
+	return shouldMeasure() &&
+	        isConnected();
+}
+
+bool REIXSSampleMotor::shouldMeasure() const
+{
+	if(horizontalTranslationControl_ && normalTranslationControl_) {
+		return horizontalTranslationControl_->shouldMeasure() &&
+				normalTranslationControl_->shouldMeasure();
+	} else {
+
+		return false;
+	}
+}
+
+bool REIXSSampleMotor::canMove() const
+{
+	return shouldMove() &&
+	        isConnected();
+}
+
+bool REIXSSampleMotor::shouldMove() const
+{
+	if(horizontalTranslationControl_ && normalTranslationControl_) {
+
+		return horizontalTranslationControl_->shouldMove() &&
+		        normalTranslationControl_->shouldMove();
+	} else {
+
+		return false;
+	}
+}
+
+bool REIXSSampleMotor::canStop() const
+{
+	return shouldStop() &&
+	        isConnected();
+}
+
+bool REIXSSampleMotor::shouldStop() const
+{
+	if(horizontalTranslationControl_ && normalTranslationControl_) {
+
+		return horizontalTranslationControl_->shouldStop() &&
+		        normalTranslationControl_->shouldStop();
+	} else {
+
+		return false;
+	}
+}
+
+bool REIXSSampleMotor::validValue(double value) const
+{
+	QVector3D globalVector = createGlobalMovementVector(value);
+
+	return globalVector.x() >= REIXS_SAMPLE_HORIZONTAL_MIN &&
+	        globalVector.x() <= REIXS_SAMPLE_HORIZONTAL_MAX &&
+	        globalVector.y() >= REIXS_SAMPLE_NORMAL_MIN &&
+	        globalVector.y() <= REIXS_SAMPLE_NORMAL_MAX;
```

Comment:
when looking into the `source/util` folder, I found that we have `AMRange`, which has a function `withinRange()` for this :8ball: